sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/randomExpressions.scala

```diff
@@ -229,16 +232,20 @@ case class Uniform(min: Expression, max: Expression, seedExpression: Expression,
         if Seq(first, second).forall(integer) => IntegerType
       case (_, ShortType) | (ShortType, _)
         if Seq(first, second).forall(integer) => ShortType
+      case (_, ByteType) | (ByteType, _)
+        if Seq(first, second).forall(integer) => ByteType
       case (_, DoubleType) | (DoubleType, _) => DoubleType
       case (_, FloatType) | (FloatType, _) => FloatType
+      case (_, d: DecimalType) => d
+      case (d: DecimalType, _) => d
       case _ =>
         throw SparkException.internalError(
           s"Unexpected argument data types: ${min.dataType}, ${max.dataType}")
     }
   }
 
   private def integer(t: DataType): Boolean = t match {
-    case _: ShortType | _: IntegerType | _: LongType => true
+    case _: ByteType | _: ShortType | _: IntegerType | _: LongType => true
```

Comment:
Can you just check `IntegralType`?

Reply:
+1

Reply:
I ended up just updating this to use the `ExpectsInputTypes` trait to simplify this code.